### PR TITLE
Add containerized example and improve readme

### PR DESCRIPTION
**Pull request by hculea**
This PR makes some modifications to the README, such as:
* adding code examples, where necessary
* adding consistent punctuation/phrasing, where necessary
* adding links to documentation, where necessary

Aside from this, a containerized example, making use of the SDK, has also been created. This has the role of showcasing to the user how to use, in a working code example, the main functionality provided by the SDK, wrt manipulating items. This is relevant since, in contrast to the code examples available in the README, which only present features in isolation, the example showcases all the necessary steps to create working code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,28 +45,28 @@ const op = OnePasswordConnect({
...
 #### Interacting with Items
 
 ```typescript
-const vaultId = "abc1234gkqt849EXAMPLE";
+const myVaultId = {vault_uuid};
 
 // Create an Item
 const newItem = ItemBuilder()
-	.setVault(myVaultId)
+	.setVault(myVaultId).setCategory("LOGIN")
```
Comment (verkaufer) [inline README.md:58]: Can you put `setCategory` on its own line, please

File: example/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+This example has been created in order to illustrate the main available functionality with regard to item manipulation, in the JavaScript Connect SDK.
+Given a (secret) string, this example will create an item, add it to given vault, retrieve it and, eventually, remove it.
+
+## Prerequisites
+
+In order to be able to run this example, one will need:
+
+* [Docker](https://docs.docker.com/install/), installed and running
+* an [1Password Connect](https://support.1password.com/secrets-automation/#step-2-deploy-a-1password-connect-server) server, hosted on your infrastructure
```
Comment (verkaufer) [inline example/README.md:11]:
```suggestion
* a [1Password Connect](https://support.1password.com/secrets-automation/#step-2-deploy-a-1password-connect-server) server, hosted on your infrastructure
```

File: example/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# JavaScript Connect SDK Example
+
+This example has been created in order to illustrate the main available functionality with regard to item manipulation, in the JavaScript Connect SDK.
```
Comment (verkaufer) [inline example/README.md:3]:
```suggestion
This example demonstrates how to create, read, and modify items available to your 1Password Connect deployment using the JavaScript Connect SDK.
```

File: example/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+Build the JavaScript docker demo
+```
+ docker build -t js-connect-sdk-example .
+```
+
+Run the docker demo with the required fields passed as environment variables
+```
+docker run -it -e OP_CONNECT_TOKEN=<YOUR_CONNECT_TOKEN> -e OP_VAULT=<YOUR_VAULT_ID> /
+    -e OP_CONNECT_HOST=<YOUR_CONNECT_HOST> -e SECRET_STRING=<ANY_RANDOM_STRING> /
+    js-connect-sdk-example
```
Comment (verkaufer) [inline example/README.md:26]:
```suggestion
docker run -it \
    -e OP_CONNECT_TOKEN=<YOUR_CONNECT_TOKEN> \
    -e OP_VAULT=<YOUR_VAULT_ID> \
    -e OP_CONNECT_HOST=<YOUR_CONNECT_HOST> \ 
    -e SECRET_STRING=<ANY_RANDOM_STRING> \
    js-connect-sdk-example
```

Let's space this out so it's easier to read and modify in the future.

File: example/index.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
+import { OnePasswordConnect, ItemBuilder } from "@1password/connect";
+import * as readline from "readline-sync"
+import steps from steps
+
+const steps = {
+    "intro": "\nHello from 1Password! In order to exemplify the end-to-end process of creating, posting, retrieving and, eventually, deleting an item, the following steps are taken: \n",
```
Comment (verkaufer) [inline example/index.js:6]:
```suggestion
    "intro": "\nHello from 1Password! In order to demonstrate creating, editing, retrieving and, eventually, deleting an item, the following steps are taken: \n",
```

**Review by verkaufer — CHANGES_REQUESTED** [review]

File: example/index.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import { OnePasswordConnect, ItemBuilder } from "@1password/connect";
+import * as readline from "readline-sync"
+import steps from steps
```
Comment (verkaufer) [inline example/index.js:3]: I'm getting an error when running the container. Was this accidentally left behind after removing a file?

File: example/README.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+Run the docker demo with the required fields passed as environment variables
+```
+docker run -it \
+    -e OP_CONNECT_TOKEN=<YOUR_CONNECT_TOKEN> \
+    -e OP_VAULT=<YOUR_VAULT_ID> \
+    -e OP_CONNECT_HOST=<YOUR_CONNECT_HOST> \ 
+    -e SECRET_STRING=<ANY_RANDOM_STRING> \
+    js-connect-sdk-example
+```
```
Comment (verkaufer) [inline example/README.md:30]: Syntax change that I neglected earlier:

```suggestion
docker run \
--env OP_CONNECT_TOKEN=<YOUR_CONNECT_TOKEN> \
--env OP_VAULT=<YOUR_VAULT_ID> \
--env OP_CONNECT_HOST=<YOUR_CONNECT_HOST> \ 
--env SECRET_STRING=<ANY_RANDOM_STRING> \
-it js-connect-sdk-example
```
```

File: example/index.js (hunk 6 above)
Comment (hculea) [reply example/index.js:3]: This line should definitely not be here. I'll investigate, to make sure there aren't any other remains of older versions of the example. Thanks for spotting this!

Comment (hculea) [reply example/index.js:3]: This seems to be the only issue, removing this line seems to be fixing it, for me. Let me know if it's the same case for you. 😀

**Review by verkaufer — APPROVED** [review]
Confirmed the example works. Thanks for adding this!

note: the below force pushes were necessary to fix earlier unsigned commits.

**Pull request merged**